src/components/v5/frame/PageLayout/PageLayout.tsx

```diff
@@ -0,0 +1,72 @@
+import React, { FC, PropsWithChildren, useEffect, useRef } from 'react';
+import { useTablet } from '~hooks';
+
+import NavigationSidebar from '../NavigationSidebar';
+import { PageLayoutProps } from './types';
+
+const displayName = 'v5.frame.PageLayout';
+
+const PageLayout: FC<PropsWithChildren<PageLayoutProps>> = ({
+  navigationSidebarProps,
+  headerContent,
+  topContent,
+  children,
+}) => {
+  const wrapperRef = useRef<HTMLDivElement | null>(null);
+  const topContentWrapperRef = useRef<HTMLDivElement | null>(null);
+  const isTablet = useTablet();
+
+  useEffect(() => {
+    if (!topContentWrapperRef.current || !wrapperRef.current) {
+      return;
+    }
+
+    const { height } = topContentWrapperRef.current.getBoundingClientRect();
+
+    wrapperRef.current.style.setProperty('--top-content-height', `${height}px`);
```

Comment:
Shouldn't you add a ResizeObserver here so when content changes height is updated as well?